data-runtime/src/main/java/io/micronaut/data/runtime/operations/internal/AbstractCascadeOperations.java

```diff
@@ -102,6 +102,10 @@ protected <T> void cascade(AnnotationMetadata annotationMetadata, Class<?> repos
                     case MANY_TO_MANY:
                         final PersistentAssociationPath inverse = association.getInversePathSide().orElse(null);
                         Iterable<Object> children = (Iterable<Object>) association.getProperty().get(entity);
+                        if (children != null) {
```

Comment:
Faced this issue converting `jdbc-example-records-java`from `H2` to `ORACLE` and it saves cascade data differently so needed to add this fix too.